### PR TITLE
Introduce visionOS Platform

### PR DESCRIPTION
**Pull request by artemcm**
Companion change to https://github.com/apple/swift/pull/72834

### CODE REVIEW

File: Sources/SwiftDriver/Utilities/Triple.swift [hunk 1]
```diff
@@ -1601,7 +1609,8 @@ extension Triple {
       if version.major < 10 {
         return nil
       }
-
+    case .visionos:
+      fallthrough
```
Comment (CodaFi) [inline Sources/SwiftDriver/Utilities/Triple.swift:1613]: Nit: this can be merged into the case below

**Review by CodaFi — APPROVED** [review]

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test Windows platform

**Pull request merged**